lib/ephy-sync-utils.c

```diff
@@ -226,3 +228,48 @@ ephy_sync_utils_get_random_sync_id (void)
 
   return id;
 }
+
+void
+ephy_sync_utils_set_device_id (const char *id)
+{
+  g_return_if_fail (id);
+
+  g_settings_set_string (EPHY_SETTINGS_SYNC, EPHY_PREFS_SYNC_DEVICE_ID, id);
+}
+
+char *
+ephy_sync_utils_get_device_id (void)
+{
+  char *id;
+
+  id = g_settings_get_string (EPHY_SETTINGS_SYNC, EPHY_PREFS_SYNC_DEVICE_ID);
+  if (!g_strcmp0 (id, "")) {
+    g_free (id);
+    id = ephy_sync_utils_get_random_sync_id ();
+  }
+
+  return id;
+}
+
+void
+ephy_sync_utils_set_device_name (const char *name)
+{
+  g_return_if_fail (name);
+
+  g_settings_set_string (EPHY_SETTINGS_SYNC, EPHY_PREFS_SYNC_DEVICE_NAME, name);
+}
+
+char *
+ephy_sync_utils_get_device_name (void)
+{
+  char *name;
+
+  name = g_settings_get_string (EPHY_SETTINGS_SYNC, EPHY_PREFS_SYNC_DEVICE_NAME);
+  if (!g_strcmp0 (name, "")) {
+    g_free (name);
+    name = g_strdup_printf ("%s's Epiphany on %s",
```

Comment:
Let's use "GNOME Web" for the device name.

Normally we say only "Web" in the user interface. But since this is going to be displayed in Firefox, I think that would be too confusing. "GNOME Web" will disambiguate without exposing the Epiphany codename to users.

It's best to continue using "Epiphany" everywhere that's not user-visible.